dbt/adapters/redshift/connections.py

```diff
@@ -253,9 +254,8 @@ def cancel(self, connection: Connection):
             raise
 
         sql = f"select pg_terminate_backend({pid})"
-        _, cursor = self.add_query(sql)
```

Comment:
This probably started throwing an error because Redshift may not let you execute `pg_terminate_backend` as part of a multi-statement execution. We started seeing issues with how multi-statement executions within transactions were handled in `psycopg2` versus `redshift-connector`. Again it came down to `psycopg2` not being PEP-249 compliant whereas `redshift-connector` is. Unfortunately we built each adapter around `psycopg2`, which leaves some artifacts like this one to work out.